drivers/scale_host.go

```diff
@@ -245,6 +305,9 @@ func (s *ScaleHostDriver) Execute(conf interface{}, apiClient *client.RancherCli
 		for _, host := range hostScalingGroup {
 			state := host.State
 			if state == "inactive" || state == "deactivating" || state == "reconnecting" || state == "disconnected" {
+				if deleteCount >= amount {
+					return http.StatusBadRequest, fmt.Errorf("Cannot scale down exceed amount")
+				}
 				badHosts[host.Id] = true
 				log.Infof("Deleting host %s with priority because of bad state: %s", host.Id, host.State)
 				code, err := deleteHost(host.Id, apiClient)
```

Comment:
delete hosts in for range hostScalingGroup, because, those hosts are in bad states. Delete hosts in if deleteOption == "mostRecent", because the # of bad state hosts < amount, which is the target number for scaling down, so, we still have (amount - deleteCount) of hosts need to be deleted